Source/Utils/WarpXUtil.cpp

```diff
@@ -214,6 +214,78 @@ WarpXParser makeParser (std::string const& parse_function, std::vector<std::stri
     return parser;
 }
 
+void CheckGriddingForRZSpectral ()
+{
+#if (defined WARPX_DIM_RZ) && (defined WARPX_USE_PSATD)
+// When using the RZ spectral solver, the Hankel transform cannot be
+// divided among multiple blocks. Each block must extend over the
+// entire radial extent.
+// The grid can be divided up along z, but the number of blocks
+// must be >= the number of processors.
+
+    int max_level;
+    Vector<int> n_cell(AMREX_SPACEDIM, -1);
+
+    ParmParse pp_amr("amr");
+
+    pp_amr.get("max_level",max_level);
+    pp_amr.getarr("n_cell",n_cell,0,AMREX_SPACEDIM);
+
+    Vector<int> blocking_factor_x(max_level+1);
+    Vector<int> max_grid_size_x(max_level+1);
+
+    // Set the radial block size to be equal to the radial grid size.
+    blocking_factor_x[0] = n_cell[0];
+    max_grid_size_x[0] = n_cell[0];
+
+    for (int lev=1 ; lev <= max_level ; lev++) {
+        // This assumes a refinement ratio of 2
+        blocking_factor_x[lev] = blocking_factor_x[lev-1]/2;
+        max_grid_size_x[lev] = max_grid_size_x[lev-1]/2;
+    }
+
+    pp_amr.addarr("blocking_factor_x", blocking_factor_x);
+    pp_amr.addarr("max_grid_size_x", max_grid_size_x);
+
+    // Adjust the longitudinal block sizes, making sure that there are
+    // more blocks than processors.
+    int nprocs = ParallelDescriptor::NProcs();
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(n_cell[1] > nprocs,
+                                     "With RZ spectral, the number of z cells must be greater than the number of processors.");
```

Comment:
```suggestion
    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(n_cell[1]/warpx.blockingFactor(0) >= nprocs,
                                     "With RZ spectral, the number of z cells must be greater than the number of processors.");
```

Reply:
Why divide n_cell by warpx.blockingFactor(0)?

Reply:
In the extreme case, are you allowing up to 1 cell per proc? If so, I don't think AMReX will let us do that, as 1 box has at least `blocking_factor` cells. B ut maybe I am misunderstanding.

Reply:
This does need to be fixed. I did some tests and WarpX worked with a blocking factor of 2 (2 cells per node) but not 1. But as this is written, it only works for n_cell[1] = power of 2. I'll fix this.

The assert makes sure that there are enough cells for the processors. The code below modifies blocking_factor_y so that there will be at least one block per processor.